clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/description/manifest/KubernetesManifest.java

```diff
@@ -242,16 +242,11 @@ public static String getFullResourceName(KubernetesKind kind, String name) {
   }
 
   public boolean isNewerThanObservedGeneration() {
-    try {
-      Long generation = (Long) getMetadata().get("generation");
-      Long observedGeneration = ((Map<String, Long>) getStatus()).get("observedGeneration");
-      if (observedGeneration == null || (generation != null && generation > observedGeneration)) {
-        return false;
-      }
-    } catch (ClassCastException e) {
-    }
-    return true;
+    Integer generation = (Integer) getMetadata().get("generation");
```

Comment:
it might be worth breaking parts of this up and checking the types before casting them:
```
    Object generationObj = getMetadata().get("generation");
    if (generationObject == null) {
        return false;
    } 
    if (!(generationObj instanceof Integer)) {
      throw IllegalStateException("Expected metadata.generation to be an Integer but was actually a " + generationObj.getClass())
    }
    Integer generation = (Integer) generationObj;
    Integer observedGeneration = ((Map<String, Integer>) getStatus()).get("observedGeneration");
    return !(observedGeneration == null || (generation > observedGeneration));
```
This does 2 things, in my mind:
1 - the error message in the case that it is not an integer value in that field is going to be more clear/actionable than just a "Oops couldn't cast from foo to integer"
2 - breaks outs one of the boolean conditions from that return line which currently has enough conditions that I considered writing out a truth table to be able to reason about which conditions resulted in which outcomes.

Reply:
I think this is reasonable, going to do it.

Reply:
Should be fixed up.